### PR TITLE
Packrat

### PR DESCRIPTION
**Pull request by jkamins7**
Switching R package version control to packrat instead of install_version (which is unstable and also started needing maintainance)

### CODE REVIEW

**Comment by jkamins7** [comment]
Files in the packrat directory can be ignored by reviewers.  They are there so that packrat always has the right packages to perform the restore.

**Comment by kkintaro** [comment]
Are checks failing because changes need to be uploaded to the Docker?

Best way to test this?

**Comment by jkamins7** [comment]
Probably add the docker checks to this branch, to see if the checks pass on this branch

**Comment by perifaws** [comment]
Should the following option be used?

`packrat::set_opts(vcs.ignore.src = TRUE)`

https://github.com/rstudio/packrat/issues/59#issuecomment-45474984

**Comment by jkamins7** [comment]
I've personally had very bad experience with R's ability to install packages of versions other than the most recent version, so including source on github seems like the way to go to me.  I've also been generating the packrat files from in the docker so having these sources speeds up restore time as well (which makes docker builds easier)

**Comment by samshah** [comment]
seems like all the libraries were accidentally checked in?

**Comment by kkintaro** [comment]
Checking in the source was intentional...

**Comment by jkamins7** [comment]
The libraries are all checked in on purpose.  R doesn't really have good support for old libraries, so I'd rather have the source in the packrat than not be able to reproduce

**Comment by jkamins7** [comment]
As an example, the current install_version code broke when R 4.0.0 released and now foreign is not available.

**Comment by jkamins7** [comment]
It is possible to use packrat without all of the sources, but it worries me.  Is there a particular reason to do so?

**Comment by samshah** [comment]
got it, i don't know enough about r package management. the downsides i see are: (i) management burden of all these files/source; (ii) violating licensing of these packages (& mixing w this repo's GPLv3 license).

File: Dockerfile [hunk 1]
```diff
@@ -87,14 +87,13 @@ ENV HOME /home/app
...
-RUN Rscript packages.R
-
-# install custom packages from R/pkgs/**
-COPY local_install.R $HOME
-COPY R/pkgs $HOME/R/pkgs
-RUN Rscript local_install.R
-
+RUN Rscript -e "install.packages('packrat',repos='https://cloud.r-project.org/')"
+RUN Rscript -e "install.packages('arrow',repos='https://cloud.r-project.org/')"
+RUN Rscript -e 'arrow::install_arrow()'
```
Comment (samshah) [inline Dockerfile:92]: should be 1 `RUN` statement cojoined with `&&`. docker creates a new layer per RUN command, which is semi-expensive, so we should make them compact.

File: Dockerfile [hunk 2]
```diff
@@ -87,14 +87,13 @@ ENV HOME /home/app
...
+COPY --chown=app:app packrat $HOME/packrat
+COPY --chown=app:app  .Rprofile $HOME/.Rprofile
+COPY --chown=app:app R/pkgs $HOME/R/pkgs
+RUN Rscript -e 'packrat::restore()'
```
Comment (samshah) [inline Dockerfile:96]: move these copies before the RUN and then merge RUN command with one above

File: Dockerfile [hunk 3]
```diff
@@ -87,14 +87,13 @@ ENV HOME /home/app
...
 # TODO: use packrat (or something else) for R package management
-COPY packages.R $HOME
-RUN Rscript packages.R
-
-# install custom packages from R/pkgs/**
-COPY local_install.R $HOME
-COPY R/pkgs $HOME/R/pkgs
-RUN Rscript local_install.R
-
+RUN Rscript -e "install.packages('packrat',repos='https://cloud.r-project.org/')"
```
Comment (samshah) [inline Dockerfile:90]: set packrat (and arrow) versions? otherwise this will always pull latest

File: Dockerfile (hunk 2 above)
Comment (jkamins7) [reply Dockerfile:96]: The commands above need to be run before the copies.  Is it ok to have 2 run statements?

File: Dockerfile (hunk 3 above)
Comment (jkamins7) [reply Dockerfile:90]: packrat is deprecated, so any updates are bugfixes.  this was the only way i could get arrow to install, will think about a better solution.  (this is what arrow::install_arrow runs)

File: Dockerfile (hunk 2 above)
Comment (samshah) [reply Dockerfile:96]: yeah, 2 run statements are fine

**Pull request merged**